supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystemAsyncJobExecutionActor.scala

```diff
@@ -266,6 +291,44 @@ trait SharedFileSystemAsyncJobExecutionActor
           |""".stripMargin)
   }
 
+  /**
+    * Send the job id of the running job to the key value store.
+    *
+    * @param runningJobOption The running job.
+    */
+  private def tellKvJobId(runningJobOption: SharedFileSystemJob): Unit = {
+    val kvJobKey =
+      KvJobKey(jobDescriptor.key.call.fullyQualifiedName, jobDescriptor.key.index, jobDescriptor.key.attempt)
+    val scopedKey = ScopedKey(jobDescriptor.workflowDescriptor.id, kvJobKey, SharedFileSystemJob.JobIdKey)
+    val kvValue = Option(runningJobOption.jobId)
+    val kvPair = KvPair(scopedKey, kvValue)
+    val kvPut = KvPut(kvPair)
+    serviceRegistryActor ! kvPut
+  }
+
+  def recoverScript(job: SharedFileSystemJob): ExecutionHandle = {
+    context.become(sharedReceive(Option(job)) orElse super.receive)
+    // To avoid race conditions, check for the rc file after checking if the job is alive.
+    if (isAlive(job) || jobPaths.returnCode.exists) {
```

Comment:
ToL: I'd switch these two around on the assumption that a FS check will be easier than the `isAlive`